### PR TITLE
fix(link): fix on hover diabled text styles

### LINKED ISSUE #11392: [Bug]: Large or small disabled links revert to medium on hover
### Package

@carbon/styles

### Browser

Chrome

### Package version

1.2.0

### React version

_No response_

### Description

In regards to the link styling in the disabled state.  When setting the link size to `sm` or `lg`, hovering over a disabled link will revert the size to medium.  I believe the issue is caused by this line: 

https://github.com/carbon-design-system/carbon/blob/v11.2.0/packages/styles/scss/components/link/_link.scss#L71

### CodeSandbox example

https://codesandbox.io/s/ecstatic-buck-dqesvy?file=/index.html

### Steps to reproduce

Create a disabled large or small anchor link:
```
<a class="cds--link cds--link--disabled cds--link--lg" href="https://www.ibm.com">Link text</a>
```
or
```
<a class="cds--link cds--link--disabled cds--link--sm" href="https://www.ibm.com">Link text</a>
```

Then hover with the cursor over the link (or trigger the hover state).  On hover, the link size will revert to 'medium'

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by aledavila**
Closes #11392

#### Changelog

**Changed**

- added hover diabled styles to sm and lg links
- 
#### Testing / Reviewing

Make sure links is working as expected
Check disabled (in all sizes) + hover

### CODE REVIEW

File: packages/styles/scss/components/link/_link.scss [hunk 1]
```diff
@@ -97,11 +97,17 @@ $link-focus-text-color: custom-property.get-var(
     text-decoration: underline;
   }
 
-  .#{$prefix}--link--sm {
+  // .#{$prefix}--link--disabled {
+  //   @include type.type-style('body-compact-01');
+  // }
```
Comment (joshblack) [inline packages/styles/scss/components/link/_link.scss]:
```suggestion
```

**Review by joshblack — CHANGES_REQUESTED** [review]
Looks good! Only thing would be to remove the commented code if it's no longer needed 👀 (wasn't sure if it was still being used or not for testing)

File: packages/styles/scss/components/link/_link.scss (hunk 1 above)
Comment (aledavila) [reply packages/styles/scss/components/link/_link.scss]: Good catch! Forgot to remove that. Was testing

**Review by jnm2377 — APPROVED** [review]
LGTM pending Josh's comment :)

**Comment by aledavila** [comment]
@joshblack can recheck

**Review by joshblack — APPROVED** [review]

**Pull request merged**